related-available/938_mariabackup.sh

```diff
@@ -208,69 +250,207 @@ mariadb_restore() {
+    ret_restore=$($MARIABACKUP_BIN --copy-back --target-dir="${FULL_BACKUP_DIR}" --datadir=${MYSQL_DB_PATH} 2>&1)
     retcode2=$?
     DATEDONE2=$(date +"%s")
     let TIMETOCOMPLEATE2="$DATEDONE2 - $DATESTART2" || true # not to stop script
 
-    if [ $retcode2 -eq 0 ] ; then
-        printlogmess -n ${SCRIPTNAME} -i ${SCRIPTID} -x ${SCRIPTINDEX} -l $INFO  -e ${ERRNO[18]} -d "${DESCR[18]}" -1 "${FULL_BACKUP_DIR}" -2 $TIMETOCOMPLEATE2
-    else
+    if [ $retcode2 != 0 ] ; then
+    #    printlogmess -n ${SCRIPTNAME} -i ${SCRIPTID} -x ${SCRIPTINDEX} -l $INFO  -e ${ERRNO[18]} -d "${DESCR[18]}" -1 "${FULL_BACKUP_DIR}" -2 $TIMETOCOMPLEATE2
+    #else
         printlogmess -n ${SCRIPTNAME} -i ${SCRIPTID} -x ${SCRIPTINDEX} -l $ERROR -e ${ERRNO[19]} -d "${DESCR[19]}" -1 "${FULL_BACKUP_DIR}" -2 $TIMETOCOMPLEATE2 -3 "$ret_restore"
     fi
 
     perms=$(chown -R ${MYSQL_FILESYSTEM_USER}:${MYSQL_FILESYSTEM_GROUP} "${MYSQL_DB_PATH}")
     retcode3=$?
-    if [ $retcode3 -eq 0 ] ; then
-        printlogmess -n ${SCRIPTNAME} -i ${SCRIPTID} -x ${SCRIPTINDEX} -l $INFO  -e ${ERRNO[20]} -d "${DESCR[20]}" -1 "${FULL_BACKUP_DIR}" -2 $TIMETOCOMPLEATE2
-    else
-        printlogmess -n ${SCRIPTNAME} -i ${SCRIPTID} -x ${SCRIPTINDEX} -l $ERROR -e ${ERRNO[21]} -d "${DESCR[21]}" -1 "${FULL_BACKUP_DIR}" -2 $TIMETOCOMPLEATE2 -3 "$ret_restore"
+    if [ $retcode3 != 0 ] ; then
+    #    printlogmess -n ${SCRIPTNAME} -i ${SCRIPTID} -x ${SCRIPTINDEX} -l $INFO  -e ${ERRNO[20]} -d "${DESCR[20]}" -1 "${FULL_BACKUP_DIR}" -2 $TIMETOCOMPLEATE2
+    #else
+        printlogmess -n ${SCRIPTNAME} -i ${SCRIPTID} -x ${SCRIPTINDEX} -l $ERROR -e ${ERRNO[21]} -d "${DESCR[21]}" -1 "${FULL_BACKUP_DIR}" -2 $TIMETOCOMPLEATE2 -3 "$perms"
     fi
 
     selinux=$(restorecon -Rv "${MYSQL_DB_PATH}")
     retcode4=$?
-    if [ $retcode4 -eq 0 ] ; then
-        printlogmess -n ${SCRIPTNAME} -i ${SCRIPTID} -x ${SCRIPTINDEX} -l $INFO  -e ${ERRNO[22]} -d "${DESCR[22]}" -1 "${FULL_BACKUP_DIR}" -2 $TIMETOCOMPLEATE2
-    else
-        printlogmess -n ${SCRIPTNAME} -i ${SCRIPTID} -x ${SCRIPTINDEX} -l $ERROR -e ${ERRNO[23]} -d "${DESCR[23]}" -1 "${FULL_BACKUP_DIR}" -2 $TIMETOCOMPLEATE2 -3 "$ret_restore"
+    if [ $retcode4 != 0 ] ; then
+    #    printlogmess -n ${SCRIPTNAME} -i ${SCRIPTID} -x ${SCRIPTINDEX} -l $INFO  -e ${ERRNO[22]} -d "${DESCR[22]}" -1 "${FULL_BACKUP_DIR}" -2 $TIMETOCOMPLEATE2
+    #else
+        printlogmess -n ${SCRIPTNAME} -i ${SCRIPTID} -x ${SCRIPTINDEX} -l $ERROR -e ${ERRNO[23]} -d "${DESCR[23]}" -1 "${FULL_BACKUP_DIR}" -2 $TIMETOCOMPLEATE2 -3 "$selinux"
     fi
+}
 
+Sub_Menu(){
+#cat ${DATAFILE} |cut -f3 -d"="| sort |uniq
+# Displays a list of valid backup directorys
+
+# Set the prompt for the select command
+PS3="Type a number or 'q' to quit, or press return to main menu: "
+
+# Create a list of files to display
+#fileList=$(stat -c '%n %.19y ' /backup/mariabackup/* )
+#fileList=$( cat ${DATAFILE} |cut -f3 -d"="| sort |uniq )
+fileList=$( cat /tmp/file.txt )
```

Comment:
use mktemp

Reply:
also I do really not like menus in syscheck it is not possible to automate.
can we convert this to a two stage thing? 1. run with --list 2. run with --select